ai-ml/jupyterhub/addons.tf

```diff
@@ -0,0 +1,244 @@
+  service_account_role_arn = module.vpc_cni_irsa.iam_role_arn
+}
+
+#---------------------------------------------------------------
+# Data on EKS Kubernetes Addons
+#---------------------------------------------------------------
+# NOTE: This module will be moved to a dedicated repo and the source will be changed accordingly.
+module "kubernetes_data_addons" {
+  # Please note that local source will be replaced once the below repo is public
+  # source = "https://github.com/aws-ia/terraform-aws-kubernetes-data-addons"
+  source            = "../../workshop/modules/terraform-aws-eks-data-addons"
+  oidc_provider_arn = module.eks.oidc_provider_arn
+  #---------------------------------------------------------------
+  # JupyterHub Add-on
+  #---------------------------------------------------------------
+  jupyterhub_helm_config = {
+    values = [templatefile("${path.module}/helm-values/jupyter-values.yaml", {
+         ssl_cert_arn   = data.aws_acm_certificate.issued.arn
+         jupyterdomain  = "https://${var.acm_certificate_domain}/hub/oauth_callback"
+         authorize_url  = "https://${local.cog_domain_name}.auth.${local.region}.amazoncognito.com/oauth2/authorize"
+         token_url      = "https://${local.cog_domain_name}.auth.${local.region}.amazoncognito.com/oauth2/token"
+         userdata_url   = "https://${local.cog_domain_name}.auth.${local.region}.amazoncognito.com/oauth2/userInfo"
+         client_id      =  aws_cognito_user_pool_client.user_pool_client.id
+         client_secret  =  aws_cognito_user_pool_client.user_pool_client.client_secret
+      })]
+    }
+  }
+  #---------------------------------------------------------------
+# EFS Filesystem for private volumes per user
+#---------------------------------------------------------------
+resource "aws_efs_file_system" "efs" {
+  creation_token = "efs"
+  encrypted      = true
+  
+  tags = local.tags
+}
+
+resource "aws_efs_mount_target" "efs_mt" {
+  count = length(compact([for subnet_id, cidr_block in zipmap(module.vpc.private_subnets, module.vpc.private_subnets_cidr_blocks) : substr(cidr_block, 0, 4) == "100." ? subnet_id : null]))
+  
+  file_system_id  = aws_efs_file_system.efs.id
+  subnet_id       = "${element(compact([for subnet_id, cidr_block in zipmap(module.vpc.private_subnets, module.vpc.private_subnets_cidr_blocks) : substr(cidr_block, 0, 4) == "100." ? subnet_id : null]), count.index)}"
+  security_groups = [aws_security_group.efs.id]
+}
+
+resource "aws_security_group" "efs" {
+  name        = "${local.name}-efs"
+  description = "Allow inbound NFS traffic from private subnets of the VPC"
+  vpc_id      = module.vpc.vpc_id
+
+  ingress {
+    description = "Allow NFS 2049/tcp"
+    cidr_blocks = module.vpc.vpc_secondary_cidr_blocks
+    from_port   = 2049
+    to_port     = 2049
+    protocol    = "tcp"
+  }
+
+  tags = local.tags
+}
+resource "kubectl_manifest" "pv" {
```

Comment:
why are these created as templated manifests? why not either:
1. Just put the manifest into the resource here and use Terraform templating
2. Use the Terraform resource for PV or PVC

Reply:
updated